src/control/lib/control/auto.go

```diff
@@ -449,3 +468,76 @@ func (req *ConfigGenerateReq) checkStorage(ctx context.Context, resp *ConfigGene
 
 	return nil
 }
+
+// checkCpus validates VOS target count and xstream helper thread count
+// recommended values and populates ioserver storage config.
+//
+// The target count should be a multiplier of the number of SSDs and typically
+// daos gets the best performance with 16x targets per I/O Server so target
+// count will be between 12 and 20.
+//
+// Validate number of targets + 1 cores are available per IO servers, not
+// usually a problem as sockets normally have at least 18 cores.
+//
+// Create helper threads for the remaining available cores, e.g. with 24 cores,
+// allocate 7 helper threads. Number of helper threads should never be more than
+// number of targets.
+func (req *ConfigGenerateReq) checkCPUs(coresPerNuma int, resp *ConfigGenerateResp) error {
+	if coresPerNuma < 1 {
+		return errors.Errorf(errInvalNumCores, coresPerNuma)
+	}
+
+	for idx, ioCfg := range resp.ConfigOut.Servers {
```

Comment:
I will generalise the algorithm, the current implementation is not general because specific special cases were requested that didn't fit into a general algorithm, the ranges selected matched what was requested.